db/client.go

```diff
@@ -44,6 +49,16 @@ func newClientModel(cli client.Client) (*clientModel, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	if cli.Public {
+		// Metadata.Valid(), and therefore json.Unmarshal(metadata) complains
```

Comment:
...but maybe we could break out metadata.Valid() into different parts, or have it more configurable.